src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs

```diff
@@ -1523,7 +1556,7 @@ void AggregateEntityResourceVaultInternal(long entityId, long resourceEntityId,
             rowsInserted += await writeHelper.CopyResourceEntitySupplyHistory(resourceEntitySupplyHistoryToAdd, token);
             rowsInserted += await writeHelper.CopyNonFungibleDataSchemaHistory(nonFungibleSchemaHistoryToAdd, token);
             rowsInserted += await writeHelper.CopyKeyValueStoreSchemaHistory(keyValueStoreSchemaHistoryToAdd, token);
-            rowsInserted += await writeHelper.CopyResourceOwners(resourceOwnersToAdd.Select(x => x.Value).ToList(), token);
+            rowsInserted += await writeHelper.CopyResourceOwners(resourceOwnersToAdd, token);
```

Comment:
It seems way overcomplicated to me. I guess I'd suggest to:
- keep track of `0` balance as if it was a regular value - it will simplify entire codebase by quite some margin,
- on the database side use MERGE with INSERT / UPDATE or DELETE clause while merging the temp table with the target table

Reply:
Think:
```sql
MERGE INTO resource_owners ro
USING tmp_resource_owners tmp
ON ro.entity_id = tmp.entity_id AND ro.resource_entity_id = tmp.resource_entity_id
WHEN MATCH AND ro.from_state_version < tmp.from_state_version AND tmp.balance = 0 THEN <<delete>>
WHEN MATCH AND ro.from_state_version < tmp.from_state_version THEN <<update>>
WHEN MATCH DO NOTHING
WHEN NOT MATCHED THEN <<insert>>
```